griptape/tasks/actions_subtask.py

```diff
@@ -0,0 +1,311 @@
+from __future__ import annotations
+import json
+import re
+from typing import Optional, TYPE_CHECKING, Callable
+import schema
+from attr import define, field
+from jsonschema.exceptions import ValidationError
+from jsonschema.validators import validate
+from schema import Schema, Literal
+from griptape import utils
+from griptape.utils import remove_null_values_in_dict_recursively
+from griptape.mixins import ActionSubtaskOriginMixin
+from griptape.tasks import BaseTextInputTask, BaseTask
+from griptape.artifacts import BaseArtifact, ErrorArtifact, TextArtifact, ListArtifact
+from griptape.events import StartActionSubtaskEvent, FinishActionSubtaskEvent
+
+if TYPE_CHECKING:
+    from griptape.memory import TaskMemory
+    from griptape.tools import BaseTool
+
+
+@define(kw_only=True)
+class Action:
+    output_label: str = field()
+    name: str = field()
+    path: Optional[str] = field(default=None)
+    input: dict = field()
+    tool: Optional[BaseTool] = field(default=None)
```

Comment:
Should this be moved inside of `ActionsSubtask` similar to how we have `PromptStack.Input`?